File: KotonohaAssistant.Web/kotonohaassistant.web.client/src/types/conversationResult.ts
import { Kotonoha } from "./sisterType";

export interface ConversationFunction {
    name: string;
    arguments: {
        [key: string]: unknown
    };
    result: string;
}

export interface ConversationResult {
    message: string;
    sister: Kotonoha;
    functions?: ConversationFunction[]
}

File: KotonohaAssistant.AI/Services/ConversationResult.cs
using KotonohaAssistant.Core;

namespace KotonohaAssistant.AI.Services;

public class ConversationResult
{
    public required string Message { get; set; }
    public required Kotonoha Sister { get; set; }
    public required Emotion Emotion { get; set; }
    public List<ConversationFunction>? Functions { get; set; }
}

public class ConversationFunction
{
    public required string Name { get; set; }
    public required IDictionary<string, object> Arguments { get; set; }
    public required string Result { get; set; }
}